tests/base/src/test/java/org/wildfly/security/auth/realm/DistributedSecurityRealmTest.java

```diff
@@ -126,7 +136,30 @@ public void testBadEvidence() throws Exception {
         identity.dispose();
     }
 
+    @Test(expected = RealmUnavailableException.class)
+    public void testRealmUnavailable() throws Exception {
+        realm = new DistributedSecurityRealm(realm1, realm2, ldapRealm, realm3);
 
+        realm.getRealmIdentity(new NamePrincipal("user3"));
+    }
+
+    @Test(expected = RealmUnavailableException.class)
+    public void testRealmUnavailableIgnoreFalse() throws Exception {
+        boolean ignoreUnavailableRealms = false;
+        realm = new DistributedSecurityRealm(ignoreUnavailableRealms, null, realm1, realm2, ldapRealm, realm3);
+
+        realm.getRealmIdentity(new NamePrincipal("user3"));
+    }
+
+    @Test
+    public void testIgnoreRealmUnavailable() throws Exception {
```

Comment:
Can you also add a test with a consumer please? Like in case of `FailoverRealmTest`, see https://github.com/wildfly-security/wildfly-elytron/blob/febc8a60e7cd34d376929e20129f0b919e2f11c1/auth/realm/base/src/test/java/org/wildfly/security/auth/realm/FailoverRealmTest.java#L58 for example.